--- origsrc/cups-1.4.6/scripting/perl/Makefile.PL	2002-08-19 15:43:03.000000000 -0500
+++ src/cups-1.4.6/scripting/perl/Makefile.PL	2011-01-29 22:31:56.452434900 -0600
@@ -8,10 +8,10 @@ WriteMakefile(
     ($] >= 5.005 ?    ## Add these new keywords supported since 5.005
       (ABSTRACT_FROM => 'CUPS.pm', # retrieve abstract from module
        AUTHOR     => 'A. U. Thor <a.u.thor@a.galaxy.far.far.away>') : ()),
-    'LIBS'		=> ['-lcups '], # e.g., '-lm'
+    'LIBS'		=> ['-L../../cups -lcups '], # e.g., '-lm'
     'DEFINE'		=> '', # e.g., '-DHAVE_SOMETHING'
 	# Insert -I. if you add *.h files later:
-    'INC'		=> '', # e.g., '-I/usr/include/other'
+    'INC'		=> '-I../..', # e.g., '-I/usr/include/other'
 	# Un-comment this if you add C files to link with later:
     # 'OBJECT'		=> '$(O_FILES)', # link all the C files too
 );
